apps/src/aichat/types.ts

```diff
@@ -146,3 +146,37 @@ export interface LevelAichatSettings {
 
 // The type of save action being performed (customization update, publish, model card save, etc).
 export type SaveType = 'updateChatbot' | 'publishModelCard' | 'saveModelCard';
+
+/** Response structure for the detect toxicity API */
+export interface DetectToxicityResponse {
+  flaggedFields: FlaggedField[];
+}
+
+export interface FlaggedField {
+  field: keyof AiCustomizations;
+  toxicity: {
+    text: string;
+    blockedBy: SafetyService;
+    details: BlocklistDetails | WebPurifyDetails | ComprehendDetails;
+  };
+}
+
+type SafetyService = 'blocklist' | 'webpurify' | 'comprehend';
+
+interface BlocklistDetails {
```

Comment:
`BlocklistDetails` seems very similar to `WebPurifyDetails` except that `type` is always profanity.
Would it make sense to have fields more similar and instead of `blockedWord`, we could use `content`? And `content` could be an array of strings if you end up returning all flagged words?

Reply:
SGTM!